### ISSUE TITLE
Inquiry on how to run "train_finetune_videodiffusion.py"

### ISSUE THREAD

**Issue by yahyunee** [XuanhaoLiu/EEG2Video#16]
I am currently working on running the `train_finetune_videodiffusion.py` script, but I have encountered a few issues that I would greatly appreciate your assistance with. I am hoping you could help clarify some points.

1. `video_path` and `prompt` in `all_40_video.yaml` 
 In the `all_40_video.yaml` configuration file, I noticed there are references to `video_path` and `prompt` for the training and validation data. However, I am unsure where to find these paths. I have the SEED-DV dataset on hand and have run the `EEG_preprocessing/segment_raw_signals_200Hz.py` script, but I did not receive anything resembling the format "data/panda01.mp4" or "a panda is eating bamboo". Could you kindly advise where I can find or generate these data? 
Should I be make a code to segment the videos and BLIP captions from the provided SEED-DV dataset, which are currently named something like 1st_10min.mp4? Additionally, I have checked all the files within the BLIP-caption folder but was unable to find a prompt like "a panda is eating bamboo". I am curious to know where the video paths and prompts for the training and validation data came from.

2. Use of `TuneMultiVideoDataset`
 I noticed that the `TuneMultiVideoDataset` class is not defined within `dataset.py`, so I created one-though not sure if I did it correctly. In line 160 of the script, the `train_dataset` is set as an instance of `TuneMultiVideoDataset`. Should I change this to `TuneAVideoDataset` instead? I would appreciate any guidance on this matter.

3.`video_path` and `video_text` paths
Regarding the `video_path` and `video_text` paths in `train_finetune_videodiffusion.py`, I have set video_path to 'SEED-DV/Video/' and video_text to 'SEED-DV/Video/BLIP-caption'. Could you confirm if this is the correct setup, or if there is a different path that I should be using?

I am really looking forward to running the EEG2Video pipeline in my setup, and I sincerely hope you can provide more details to help me move forward. I notice that some of my questions might be hard or tedious for you to explain in text. I'm opened to any zoom calls-ofc only if you are also okay with that-so just let me know which way you prefer. Thank you once again for your amazing work and for your time in assisting with this.

Best regards,
Lucy

**Comment by XuanhaoLiu** [comment]
We have released a new version of our codebase, Please check it out in EEG2Video folder's EEG2Video_New. 

> but I did not receive anything resembling the format "data/panda01.mp4" or "a panda is eating bamboo". Could you kindly advise where I can find or generate these data?

You should write a script to segment the whole video blocks into small video clips. That is, segment every 2-second video clips and restore them in another folder.

> Should I change this to TuneAVideoDataset instead?

Yes, actually you can change the training dataset whatever you want. Like in our paper, we chose 10 classes, 20 classes, 30 classes. These subdatasets are different from each other, you can change them.

> Could you confirm if this is the correct setup, or if there is a different path that I should be using?

Sorry, I do not remember that clear. But fine-tuning the video diffusion module is the same as the Tune-A-Video.

> I'm opened to any zoom calls-ofc only if you are also okay with that-so just let me know which way you prefer.

Sure, you can contact me and send me the zoom meeting link at haogram_sjtu@sjtu.edu.cn

**Comment by yahyunee** [comment]
Hi! Thank you for the reply! So, as far as I understand,
> "script to segment the whole video blocks into small video clips" 

Does not exist in the current updated code right?

Also, what should `video_path = 'dataset/SEED-DV/output/block1` include? videos from one video file(ex. 1st_10min.mp4)?

**Comment by yahyunee** [comment]
Do you have any tips for generating 200 video clips from 1 video? I'm wondering how to name each videos. Did you just named it sequentially? 1.mp4, 2.mp4, 3.mp4 ...?

**Comment by XuanhaoLiu** [comment]
> Does not exist in the current updated code right?

No, they are not exist in our codebase. I may find my old code and update it.

> Did you just named it sequentially? 1.mp4, 2.mp4, 3.mp4 ...?

Yes, we just name it sequentially.

**Comment by XuanhaoLiu** [comment]
We have just upload the extract_gif.py in the EEG2Video file, please check it out.

Hope it can help you!
